src/components/CreateAddress/AddressGenerator.jsx

```diff
@@ -37,10 +38,14 @@ class AddressGenerator extends React.Component {
   };
 
   publicKeyCount = () => {
-    const { publicKeyImporters } = this.props;
-    return Object.values(publicKeyImporters).filter(
-      (publicKeyImporter) => publicKeyImporter.finalized
-    ).length;
+    const { publicKeyImporters, addressType } = this.props;
+    return Object.values(publicKeyImporters).filter((publicKeyImporter) => {
+      const publicKeyError = validatePublicKey(
+        publicKeyImporter.publicKey,
+        addressType
+      );
+      return publicKeyImporter.finalized && !publicKeyError;
+    }).length;
```

Comment:
this is just a personal preference/styling thing, but it might look cleaner to destructure in the filter callback 

```suggestion
    return Object.values(publicKeyImporters).filter(({ publicKey, finalized }) => 
      finalized && !validatePublicKey(publicKey, addressType)
    ).length;
```
(I haven't tested this so please confirm it works)